### ISSUE TITLE
Path completion don't escape special characters correctly

### ISSUE THREAD

**Issue by crides** [nushell/nushell#11789]
### Describe the bug

https://github.com/nushell/nushell/blob/c2992d5d8b8541b64e9fc1440ed2917fc9b757f7/crates/nu-cli/src/completions/completion_common.rs#L157-L158

This doesn't take into account `{}` which will mess up parsing. I'm also not sure why `()` is being ignored for directories.

`#` is not actually required to be in the list, with regards to the current parser. Currently, `#` will only be parsed as a comment if there are spaces surrounding both sides, which means `' '` will take care of that

### How to reproduce

1. `touch`/`mkdir` a bunch of files/directories like `(a)`, `{a}`, `a}{`, `a#`, `#a`
2. use them in completions e.g. `ls <tab>`

### Expected behavior

correctly escape the characters

### Configuration

| key                | value                                                                                  |
| ------------------ | -------------------------------------------------------------------------------------- |
| version            | 0.89.1                                                                                 |
| branch             | dev                                                                                    |
| commit_hash        | f37307bc81c7820e293f4e0a7d956cfca0a709f2                                               |
| build_os           | linux-x86_64                                                                           |
| build_target       | x86_64-unknown-linux-gnu                                                               |
| rust_version       | rustc 1.73.0 (cc66ad468 2023-10-03)                                                    |
| rust_channel       | 1.73.0-x86_64-unknown-linux-gnu                                                        |
| cargo_version      | cargo 1.73.0 (9c4383fb5 2023-08-26)                                                    |
| build_time         | 2024-02-01 20:24:06 -08:00                                                             |
| build_rust_channel | debug                                                                                  |
| allocator          | mimalloc                                                                               |
| features           | default, sqlite, trash, which, zip                                                     |
| installed_plugins  | sound, sound beep, sound make, sound meta, sound meta list, sound meta set, sound play |

### Additional context

Related: #11766

**Comment by crides** [comment]
I forgot about this, and just ran into it today, but <code>`</code> (the backtick) also needs to be escaped...

**Comment by CabalCrow** [comment]
The path autocompletion should probably be changed to use ``"`` quotes & only escape ``"`` and ``\`` characters in path names & that should be fine?

**Comment by crides** [comment]
Yes, something like that. Or use `'` or the new `r#''#` raw strings (although proper raw string seem a bit excessive)

**Comment by CabalCrow** [comment]
> Yes, something like that. Or use `'` or the new `r#''#` raw strings (although proper raw string seem a bit excessive)

The problem with raw strings is that now you need to go extra steps to sanitize input. You will need to check the path string for '#es and apply the correct number of #es on both sides. It is much simpler to just quote via ". Quoting via ' is also not a good idea since you won't be able to properly escape ' in a path. " works best nice it allows you to simple escape all \es and "es and then you are done.
